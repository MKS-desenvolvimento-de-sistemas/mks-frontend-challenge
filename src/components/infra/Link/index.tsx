import NextLink from "next/link";
import { ReactNode } from "react";

interface ILinkProps {
  href: string;
  title: string;
  target?: "blank" | "self" | "parent" | "top";
  children?: ReactNode | ReactNode[];
}

const Link = ({ href, title, target = "self", children }: ILinkProps) => {
  return (
    <NextLink href={href}>
      <a title={title} target={`_${target}`}>
        {children}
      </a>
    </NextLink>
  );
};

export default Link;
