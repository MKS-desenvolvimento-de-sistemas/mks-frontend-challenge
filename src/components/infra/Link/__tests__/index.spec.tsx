import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import Link from "../";
import renderer from "react-test-renderer";

describe("Link Component", () => {
  it("should render the link correctly, with target _self by default", () => {
    render(
      <Link href="/" title="TESTE">
        TESTE
      </Link>
    );

    const link = screen.getByTestId("link");

    expect(link).toHaveTextContent("TESTE");
    expect(link.getAttribute("href")).toEqual("/");
    expect(link.getAttribute("title")).toEqual("TESTE");
    expect(link.getAttribute("target")).toEqual("_self");
  });

  it("should render the link correctly, with target set by props", () => {
    render(
      <Link href="/" title="BLANK-TESTE" target="blank">
        BLANK-TESTE
      </Link>
    );

    const link = screen.getByTestId("link");

    expect(link).toHaveTextContent("BLANK-TESTE");
    expect(link.getAttribute("href")).toEqual("/");
    expect(link.getAttribute("title")).toEqual("BLANK-TESTE");
    expect(link.getAttribute("target")).toEqual("_blank");
  });

  it("should match the snapshot - target default", () => {
    const tree = renderer
      .create(
        <Link href="/" title="TESTE">
          TESTE
        </Link>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should match the snapshot - set target", () => {
    const tree = renderer
      .create(
        <Link href="/" title="BLANK-TESTE" target="blank">
          BLANK-TESTE
        </Link>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
