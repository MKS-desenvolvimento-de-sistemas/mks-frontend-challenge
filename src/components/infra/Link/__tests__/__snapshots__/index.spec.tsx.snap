// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Component should match the snapshot - set target 1`] = `
<a
  data-testid="link"
  href="/"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onTouchStart={[Function]}
  target="_blank"
  title="BLANK-TESTE"
>
  BLANK-TESTE
</a>
`;

exports[`Link Component should match the snapshot - target default 1`] = `
<a
  data-testid="link"
  href="/"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onTouchStart={[Function]}
  target="_self"
  title="TESTE"
>
  TESTE
</a>
`;
